### PR TITLE
Update/crm api v3 - contact merge audit removed

### PR DESCRIPTION
**Pull request by fivetran-jamie**
**Are you a current Fivetran customer?** 

internal 

**What change(s) does this PR introduce?** 

aligns with recent [changes](https://fivetran.com/docs/applications/hubspot/changelog#november2022) to the hubspot API 
- adds the new `property_hs_calculated_merged_vids` field to the contact model/macro and its seed data
- explicitly casts all timestamp fields as timestamps
- adds BigQuery-specific logic since the above connector change is blocked on BQ. Thus, they will have to use the old behavior, but we want to make it so that they can switch to the new way if they want (by setting `hubspot_contact_merge_audit_enabled` to FALSE.

**Did you update the CHANGELOG?** 

- [x] Yes

**Does this PR introduce a breaking change?**

- [x] Yes (please provide breaking change details below.)
- [ ] No  (please provide an explanation as to how the change is non-breaking below.)

default behavior is different for most users (though they _should_ end up with the same result...so maybe not?)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [x] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [x] Yes, Issue/Feature https://github.com/fivetran/dbt_hubspot_source/issues/97 + https://github.com/fivetran/dbt_hubspot/issues/97
- [ ] No 

**How did you test the PR changes?** 

 

- [x] Buildkite  
- [x] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [x] BigQuery
- [x] Redshift
- [x] Snowflake
- [x] Postgres
- [x] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
😩 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

**Comment by fivetran-jamie** [comment]
will regen docs before merging

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-jamie just a few minor requests and callouts. Let me know if you have any other questions!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+# dbt_hubspot_source v0.9.0 or v0.8.1 - breaking or na
+
+## 🎉 Features
+[PR #98](https://github.com/fivetran/dbt_hubspot_source/pull/98) applies the following changes:
+- Ensures that all timestamp fields are explicitly cast as timestamps (without timezones).
+- Updates logic around the recently deprecated `CONTACT_MERGE_AUDIT` table.
+  - In [November 2022](https://fivetran.com/docs/applications/hubspot/changelog#november2022), the Fivetran Hubspot connector switched to v3 of the Hubspot CRM API, which deprecated the `CONTACT_MERGE_AUDIT` table and stored merged contacts in the `property_hs_calculated_merged_vids` column in the `CONTACT` table. **This has not been rolled out to BigQuery warehouses yet.**
```
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply CHANGELOG.md:7]: bumped drew but he's ooo till monday

File: models/src_hubspot.yml [hunk 2]
```diff
@@ -164,7 +164,7 @@ sources:
         identifier: "{{ var('hubspot_contact_merge_audit_identifier', 'contact_merge_audit')}}"
         description: Each record contains a contact merge event and the contacts effected by the merge.
         config:
-          enabled: "{{ var('hubspot_marketing_enabled', true) and var('hubspot_contact_merge_audit_enabled', false) }}"
+          enabled: "{{ var('hubspot_marketing_enabled', true) and var('hubspot_contact_merge_audit_enabled', target.type == 'bigquery') }}"
```
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply models/src_hubspot.yml:168]: just gotta return a boolean!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,17 @@
+# dbt_hubspot_source v0.9.0 or v0.8.1 - breaking or na
```
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply CHANGELOG.md:1]: updating

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,17 @@
...
+- Ensures that all timestamp fields are explicitly cast as timestamps (without timezones).
+- Updates logic around the recently deprecated `CONTACT_MERGE_AUDIT` table.
+  - In [November 2022](https://fivetran.com/docs/applications/hubspot/changelog#november2022), the Fivetran Hubspot connector switched to v3 of the Hubspot CRM API, which deprecated the `CONTACT_MERGE_AUDIT` table and stored merged contacts in the `property_hs_calculated_merged_vids` column in the `CONTACT` table. **This has not been rolled out to BigQuery warehouses yet.**
+  - The package now brings in the new `property_hs_calculated_merged_vids` field (and removes the `property_hs_` prefix) for all customers.
+  - The package will only reference the old `CONTACT_MERGE_AUDIT` table if `hubspot_contact_merge_audit_enabled` is explicitly set to `true`, or if you are running the package on a BigQuery destination.
+    - If you are on BigQuery, but would not like to reference `CONTACT_MERGE_AUDIT`, set `hubspot_contact_merge_audit_enabled` to false. This will result in no merging of contacts (until your connector is rolled onto v3 of the Hubspot CRM API).
+  - Updates seed data to test new merging paradigm.
```
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply CHANGELOG.md:11]: cool also gonna move the `Ensures that all timestamp fields are explicitly cast as timestamps without timezone` part there as well

File: models/src_hubspot.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/src_hubspot.yml:168]: @fivetran-jamie after what we adjusted in the transform package, should we use this approach here? Not all bigquery customers will want to have `hubspot_contact_merge_audit_enabled` set as `true` by default.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-jamie this is looking great and I feel it is very close to being able to merge through. I do just have one remaining comment regarding the `hubspot_contact_merge_audit_enabled` variable and how we use it for BigQuery users. I think we need to rethink how we use this variable. We can chat during standup.

Additionally, I came across a bug when doing more rigorous testing and ended up opening PR #100 to address the issue. We can batch this with these changes. It would be great if you could review.

File: models/src_hubspot.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply models/src_hubspot.yml]: updated

**Review by fivetran-joemarkiewicz — APPROVED** [review]
Thanks for working through the updates @fivetran-jamie! After final review, these look good to go!

**Pull request merged**